hasher-matcher-actioner/hmalib/dto.py

```diff
@@ -0,0 +1,39 @@
+import datetime
+import typing as t
+
+"""
+Not enforceable because named tuples can't have multiple inheritance, but all
+DTO classes in this module should implement methods `to_dynamodb_item(self)` and
+`to_sqs_message(self)`
+"""
+
+class PDQHashRecord(t.NamedTuple):
+    """
+    Successful execution at the hasher produces this record.
+    """
+
+    content_key: str
+    content_hash: str
+    quality: int
+    timestamp: datetime.datetime  # ISO-8601 formatted
+
+    @staticmethod
+    def get_dynamodb_pk(key: str):
+        return f"c#{key}"
+
+    def to_dynamodb_item(self) -> dict:
+        return {
+            "PK": PDQHashRecord.get_dynamodb_pk(self.content_key),
+            "SK": "type:pdq",
```

Comment:
@schatten was the change from 'type#pdq' to 'type:pdq' intentional? Going to change it back in my next PR if not. Thanks!

Reply:
Nope. Totally unintentional. Nice spot. Thanks!